pkg/cloud/api/diff.go

```diff
@@ -54,10 +66,10 @@ func (r *DiffResult) add(state DiffItemState, p Path, a, b reflect.Value) {
 		State: state,
 		Path:  p,
 	}
-	if a.IsValid() {
+	if a.IsValid() && a.CanInterface() {
```

Comment:
Interface() will panic if it is called on unexported struct's fields https://pkg.go.dev/reflect#Value.Interface.
Our struct has all fields exported but I thought that still we should check that.
I revisited this function and this code is only needed for printing the diff so I changed Interface() to String(). This way we will have the same result (printing which value diff) and it will be safe for both exported and unexported fields.
I added new test for struct with unexported fields.